__tests__/implementation.test.js

```diff
@@ -292,6 +292,362 @@ test('batch endpoint (multiple requests)', async () => {
     });
 });
 
+test('batch endpoint (multiple requests) with secondaryBatchKey', async () => {
+    const config = {
+        resources: {
+            foo: {
+                isBatchResource: true,
+                docsLink: 'example.com/docs/bar',
+                batchKey: 'foo_ids',
+                newKey: 'foo_id',
+                secondaryBatchKey: 'properties',
+                secondaryNewKey: 'property',
+            },
+        },
+    };
+
+    const resources = {
+        foo: ({ foo_ids, properties, include_extra_info }) => {
+            if (_.isEqual(foo_ids, [2, 1])) {
+                expect(include_extra_info).toBe(false);
+                return Promise.resolve([
+                    { foo_id: 1, rating: 3, name: 'Burger King' },
+                    { foo_id: 2, rating: 4, name: 'In N Out' },
+                ]);
```

Comment:
do we need to support this use case? if we're only supporting the contract here https://github.com/Yelp/dataloader-codegen/pull/261#discussion_r633819405

Reply:
Hmm, if we only support the contract here [#261 (comment)](https://github.com/Yelp/dataloader-codegen/pull/261#discussion_r633819405), the contract in the original issue https://github.com/Yelp/dataloader-codegen/issues/75 will not be suppported either. I think it doesn't hurt to support more cases?

Reply:
I thought we discussed that the issue in #75 can't be supported because we can't do the data masking.

I think it's ok to support the one safer way (#261) and ask folks to switch their endpoints over if they want property merging

Reply:
@magicmark I got feedback from folks who own the endpoints we're trying to solve internally, sent message on slack. I think it's okay to support their use case since the contract is one to one correspondence between the input properties and the output properties. Please let me know if you have other thoughts